Python/pystate.c

```diff
@@ -1057,6 +1057,47 @@ _PyErr_SetInterpreterAlreadyRunning(void)
     PyErr_SetString(PyExc_InterpreterError, "interpreter already running");
 }
 
+int
+_PyInterpreterState_IsShuttingDown(PyInterpreterState *interp)
+{
+    return get_main_thread(interp) == _PyInterpreterState_MAIN_SHUTTING_DOWN;
+}
+
+int
+_PyInterpreterState_SetShuttingDown(PyInterpreterState *interp)
+{
+    void *expected = NULL;
+    // XXX Use an exchange without a comparison?
+    if (_Py_atomic_compare_exchange_ptr(&interp->threads.main,
+                                        &expected,
+                                        _PyInterpreterState_MAIN_SHUTTING_DOWN) == 0)
```

Comment:
```suggestion
    int res = _Py_atomic_compare_exchange_ptr(&interp->threads.main, &expected,
              _PyInterpreterState_MAIN_SHUTTING_DOWN);
    if (res == 0)
```
(Maybe this will improve readability. A tip :) )

Reply:
Eh, I'd prefer to keep it how it is.